.github/workflows/frontend-docs-generation.yml

```diff
@@ -1,31 +1,37 @@
 name: Generate Documentation for the frontend
 
 on:
-    push:
-        branches:
-            - '**'
-    pull_request:
-        branches:
-            - '**'
+  push:
+    branches:
+      - '**'
+  pull_request:
+    branches:
+      - '**'
 
 jobs:
-    generate-docs:
-        runs-on: ubuntu-latest
-        steps:
-            - name: Clone project
-              uses: actions/checkout@v3
+  generate-docs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone project
+        uses: actions/checkout@v4
 
-            - name: Install dependencies
-              run: yarn install
+      - name: Use Node.js 22.x
+        uses: actions/setup-node@v3
+        with:
+          node-version: 22.x
+          cache: 'npm'
 
-            - name: Generate docs in markdown
-              run: yarn run typedoc
+      - name: Install dependencies
+        run: yarn install
```

Comment:
- Ensures Yarn operates in CI mode for deterministic installs

Reply:
On it!